integration-tests/docker/chainlink.go

```diff
@@ -0,0 +1,324 @@
+			WithPollInterval(1 * time.Second),
+		Files: []tc.ContainerFile{
+			{
+				HostFilePath:      configFile.Name(),
+				ContainerFilePath: configPath,
+				FileMode:          0644,
+			},
+			{
+				HostFilePath:      secretsFile.Name(),
+				ContainerFilePath: secretsPath,
+				FileMode:          0644,
+			},
+			{
+				HostFilePath:      adminCredsFile.Name(),
+				ContainerFilePath: adminCredsPath,
+				FileMode:          0644,
+			},
+			{
+				HostFilePath:      apiCredsFile.Name(),
+				ContainerFilePath: apiCredsPath,
+				FileMode:          0644,
+			},
+		},
+	}, nil
+}
+
+type Chainlink struct {
+	prefix      string
+	container   tc.Container
+	dbContainer tc.Container
+	// export
+	Endpoint string
+}
+
+func NewChainlink(cfg any) ComponentSetupFunc {
+	// static prefix must be defined here, so later we can select all Components with this prefix
+	c := &Chainlink{prefix: "chainlink"}
+	return func(network string) (Component, error) {
+		return c.Start(network, cfg)
+	}
+}
+
+func (m *Chainlink) Prefix() string {
+	return m.prefix
+}
+
+func (m *Chainlink) Containers() []tc.Container {
+	containers := make([]tc.Container, 0)
+	return append(containers, m.dbContainer, m.container)
+}
+
+func (m *Chainlink) Stop() error {
+	if err := m.container.Terminate(context.Background()); err != nil {
+		return err
+	}
+	return m.dbContainer.Terminate(context.Background())
+}
+
+func (m *Chainlink) Start(dockerNet string, cfg any) (Component, error) {
+	nco, ok := cfg.(NodeConfigOpts)
+	if !ok {
+		return m, fmt.Errorf("cfg must be of type NodeConfigOpts")
+	}
+	networks := []string{dockerNet}
+	pgOpts := PgOpts{
+		Port:     "5432",
+		User:     "postgres",
+		Password: "test",
+		DbName:   "cl-node",
+		Networks: networks,
+	}
+	pgReq := pgContainerRequest(pgOpts)
+	dbContainer, err := tc.GenericContainer(context.Background(), tc.GenericContainerRequest{
+		ContainerRequest: pgReq,
+		Started:          true,
+	})
+	m.dbContainer = dbContainer
+	if err != nil {
+		return m, err
+	}
+
+	nodeSecrets, err := ExecuteTemplate(NodeSecretsTemplate, NodeSecretsTemplateValues{
+		PgHost: pgReq.Name,
+		PgPort: "5432",
+	})
+	if err != nil {
+		return m, err
+	}
+	clReq, err := nodeContainerRequest(m.prefix, nco, nodeSecrets, networks)
+	if err != nil {
+		return m, err
+	}
+	clC, err := tc.GenericContainer(context.Background(), tc.GenericContainerRequest{
+		ContainerRequest: *clReq,
+		Started:          true,
+	})
+	if err != nil {
+		return m, err
+	}
+	m.container = clC
+	ctName, err := clC.Name(context.Background())
+	if err != nil {
+		return m, err
+	}
+	ctName = strings.Replace(ctName, "/", "", -1)
+	clEndpoint, err := clC.Endpoint(context.Background(), "http")
+	if err != nil {
+		return m, err
+	}
+	m.Endpoint = clEndpoint
+
+	log.Info().Str("containerName", ctName).
+		Str("clEndpoint", clEndpoint).
+		Msg("Started Chainlink Node container")
+	return m, nil
+}
+
+func pgContainerRequest(o PgOpts) tc.ContainerRequest {
```

Comment:
@skudasov I suggest to make all container requests public so that developers can use it in unit/integration tests